java/vector/src/main/java/org/apache/arrow/vector/util/DecimalUtility.java

```diff
@@ -150,19 +148,32 @@ public static BigDecimal getBigDecimalFromByteBuffer(ByteBuffer bytebuf, int sta
     return new BigDecimal(unscaledValue, scale);
   }
 
+  public static byte[] getByteArrayFromArrowBuf(ArrowBuf bytebuf, int index) {
+    final byte[] value = new byte[DECIMAL_BYTE_LENGTH];
+    final int startIndex = index * DECIMAL_BYTE_LENGTH;
+    bytebuf.getBytes(startIndex, value, 0, DECIMAL_BYTE_LENGTH);
+    return value;
+  }
+
   public static void writeBigDecimalToArrowBuf(BigDecimal value, ArrowBuf bytebuf, int index) {
     final byte[] bytes = value.unscaledValue().toByteArray();
```

Comment:
Are there issues with endianness here? Java is big endian: https://docs.oracle.com/javase/7/docs/api/java/math/BigInteger.html#toByteArray() but Arrow is little endian? https://github.com/apache/arrow/blob/master/format/Layout.md#byte-order-endianness

cc @cpcloud

Reply:
Checked with @cpcloud offline. This is consistent with the C++ implementation.

Reply:
Thanks for checking on that